// box-ui-elements/es/elements/content-sidebar/activity-feed/annotations/messages.js
import { defineMessages } from 'react-intl';
var messages = defineMessages({
  annotationActivityDeleteMenuItem: {
    "id": "be.contentSidebar.activityFeed.annotationActivity.annotationActivityDeleteMenuItem",
    "defaultMessage": "Delete"
  },
  annotationActivityDeletePrompt: {
    "id": "be.contentSidebar.activityFeed.annotationActivity.annotationActivityDeletePrompt",
    "defaultMessage": "Are you sure you want to permanently delete this comment?"
  },
  annotationActivityPostedFullDateTime: {
    "id": "be.contentSidebar.activityFeed.annotationActivity.annotationActivityPostedFullDateTime",
    "defaultMessage": "{time, date, full} at {time, time, short}"
  },
  annotationActivityPageItem: {
    "id": "be.contentSidebar.activityFeed.annotationActivityPageItem",
    "defaultMessage": "Page {number}"
  },
  annotationActivityVersionLink: {
    "id": "be.contentSidebar.activityFeed.annotationActivityVersionLink",
    "defaultMessage": "Version {number}"
  },
  annotationActivityVersionUnavailable: {
    "id": "be.contentSidebar.activityFeed.annotationActivityVersionUnavailable",
    "defaultMessage": "Version Unavailable"
  }
});
export default messages;
