// box-ui-elements/es/features/unified-share-modal/constants.js
// Shared link access level constants
var ANYONE_WITH_LINK = 'peopleWithTheLink';
var ANYONE_IN_COMPANY = 'peopleInYourCompany';
var PEOPLE_IN_ITEM = 'peopleInThisItem'; // Shared link permission level constants

var CAN_VIEW_DOWNLOAD = 'canViewDownload';
var CAN_VIEW_ONLY = 'canViewOnly'; // Invitee permission level constants

var EDITOR = 'Editor';
var CO_OWNER = 'Co-owner';
var PREVIEWER = 'Previewer';
var PREVIEWER_UPLOADER = 'Previewer Uploader';
var OWNER = 'Owner';
var VIEWER = 'Viewer';
var VIEWER_UPLOADER = 'Viewer Uploader';
var UPLOADER = 'Uploader';
var COLLAB_GROUP_TYPE = 'group';
var COLLAB_USER_TYPE = 'user';
var COLLAB_PENDING_TYPE = 'pending'; // Default allowed access levels

var ALLOWED_ACCESS_LEVELS = {
  peopleInThisItem: true,
  peopleInYourCompany: true,
  peopleWithTheLink: true
};
var INVITEE_PERMISSIONS = [{
  default: false,
  text: CO_OWNER,
  value: CO_OWNER
}, {
  default: true,
  // default in the WebApp
  text: EDITOR,
  value: EDITOR
}, {
  default: false,
  text: PREVIEWER,
  value: PREVIEWER
}, {
  default: false,
  text: PREVIEWER_UPLOADER,
  value: PREVIEWER_UPLOADER
}, {
  default: false,
  text: UPLOADER,
  value: UPLOADER
}, {
  default: false,
  text: VIEWER,
  value: VIEWER
}, {
  default: false,
  text: VIEWER_UPLOADER,
  value: VIEWER_UPLOADER
}];
export { ALLOWED_ACCESS_LEVELS, ANYONE_IN_COMPANY, ANYONE_WITH_LINK, CAN_VIEW_DOWNLOAD, CAN_VIEW_ONLY, COLLAB_GROUP_TYPE, COLLAB_PENDING_TYPE, COLLAB_USER_TYPE, CO_OWNER, EDITOR, INVITEE_PERMISSIONS, OWNER, PEOPLE_IN_ITEM, PREVIEWER, PREVIEWER_UPLOADER, UPLOADER, VIEWER, VIEWER_UPLOADER };
